### ISSUE TITLE
Colorpicker Build Broken

### ISSUE THREAD

**Issue by warmans** [Promact/md2#137]
#As of today this library fails to build with the following errors:
```
ERROR in .../$$_gendir/node_modules/md2/colorpicker/colorpicker.ngfactory.ts (249,41): Property 'inputFocused' does not exist on type 'Md2Colorpicker'.
.../$$_gendir/node_modules/md2/colorpicker/colorpicker.ngfactory.ts (1301,41): Property 'isInputValidColor' is private and only accessible within class 'Md2Colorpicker'.
.../$$_gendir/node_modules/md2/colorpicker/colorpicker.ngfactory.ts (1340,9): Type 'QueryList<any>' is not assignable to type 'Portal<any>'.
  Property '_attachedHost' is missing in type 'QueryList<any>'.

```

Seems like the latest typescript updates is much stricter on typing and member visibility. TS 2.0.10 is the last version I received so I guess it's the culprit.

(Using angular CLI)

Edit: actually it seems like I've been using 2.0.10 for a while. No idea why it's suddenly complaining about this...

**Comment by mbeckenbach** [comment]
@warmans I'm having the same issue after i've updated an angular cli project to angular 4.0
Do you have any work arround for the moment?

**Comment by mbeckenbach** [comment]
@warmans typescript version 2.2.1 builds fine for me

**Comment by warmans** [comment]
@mbeckenbach I have not found a workaround yet. Upgrading to typescript 2.2.1 does not allow by build to complete. I'll probably just vendor in the md2 lib and remove the colorpicker since don't use it anyway then migrate back to the npm installed version once it's fixed :/

**Comment by mbeckenbach** [comment]
@warmans 
I had the exactly the same idea a minute ago :-)

**Comment by marius-oprea** [comment]
well not typescript 2.0.10 break the color picker, but the latest @angular/cli-1.0.0 and angular 4.0.0 final release in my case
/node_modules/md2/colorpicker/colorpicker.ngfactory.ts (1
,1): Property 'isInputValidColor' is private and only accessible within class 'Md2Colorpicker'.

**Comment by pajaroblanco** [comment]
I'm having the same problem since updating the angular-cli to the latest version 1.0.0

**Comment by hccampos** [comment]
There's this https://github.com/Promact/md2/pull/142/files

For now we just forked the project and use a custom build.

**Comment by dharmeshpipariya-zz** [comment]
We have fixed the issue with release md2@0.0.18,
comment here if you have an issue
